src/OrchardCore.Modules/OrchardCore.Menu/Migrations.cs

```diff
@@ -70,23 +70,6 @@ private static void MigrateMenuItems(List<ContentItem> menuItems)
         {
             foreach (var menuItem in menuItems)
             {
-                var linkMenuItemPart = menuItem.As<LinkMenuItemPart>();
```

Comment:
This whole method can now be removed, and also `UpdateFrom3Async`, since it's not useful anymore.

Technically, we could also get rid of all the migrations apart from the ones in `Create()`.

Reply:
I don't want to mess with migrations. There is a chance someone still need to migrate their tenants mid way. here I just want to remove obsolete code and not change the behavior. We already have lots of changes going into 2 and I don't want to complicate things even more. I'll look at the reference line when I have time

Reply:
What do you mean by "reference line"?

Reply:
@Piedone I looked at the "reference line" the code you pointed out on line 73. I think I want to leave migrations out of this PR. This PR already has lots of files changed. I would say we can create a follow up PR to cleanup old migrations where we can then carefully review the migrations related PR to ensure we don't miss anything.

Reply:
OK, let's leave those. I'd also be much more confident with better automated QA.